src/index.js

```diff
@@ -112,21 +114,18 @@ class MiniCssExtractPlugin {
   constructor(options) {
     this.options = Object.assign(
       {
-        filename: '[name].css',
+        filename: DEFAULT_FILENAME,
       },
       options
     );
     if (!this.options.chunkFilename) {
       const { filename } = this.options;
-      const hasName = filename.includes('[name]');
-      const hasId = filename.includes('[id]');
-      const hasChunkHash = filename.includes('[chunkhash]');
       // Anything changing depending on chunk is fine
-      if (hasChunkHash || hasName || hasId) {
+      if (typeof filename === 'string' && REGEXP_PLACEHOLDERS.test(filename)) {
```

Comment:
Don't use regexp , it is bad for perf

Reply:
Normally I agree but in this case it was faster than 3 `String.includes` http://jsben.ch/LMM6o

Reply:
@kevinchappell thanks for link, looks good :+1: